logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/OfferStrategy.java

```diff
@@ -50,9 +48,13 @@
     }
 
     public OfferResult evaluate(ClusterState clusterState, Protos.Offer offer) {
-        final Optional<OfferRule> decline = acceptanceRules.stream().filter(offerRule -> offerRule.rule.accepts(clusterState, offer)).limit(1).findFirst();
-        if (decline.isPresent()) {
-            return OfferResult.decline(decline.get().declineReason);
+        final List<String> complaints =
```

Comment:
A list of complaints doesn't make sense. If a machine already runs the executor, it will fail on (up to) all rules, when in reality it's only "failing" one rule. Complaints should be prioritised.